{
  try {
    ret=method.invoke(target,args);
synchronized (this) {
      completed=true;
      notify();
    }
  }
 catch (  Exception e) {
    Log.e(LOGTAG,"Failed to invoke: " + method.getName(),e);
  }
}
